### PR TITLE
Add support for python2 and python3 (no minor in version)

### LINKED ISSUE #7: Support newer python versions
When virtualenvrunner is installed it creates entry points for python 2.7, 3.4, 3.5 and 3.6 but not for python 3.7 or 3.8.

Creating entry points only for python versions which are available on the system should be considered.

### PR DESCRIPTION
**Pull request by beleki**
fix #7

### CODE REVIEW

File: src/virtualenvrunner/pythonversionrun.py [hunk 1]
```diff
@@ -20,16 +20,22 @@ def _get_run_lambda(self, python_version):
         return lambda: self._run_with_python_version(python_version)
 
     def _run_with_python_version(self, python_version):
-        self._run(pythonexe='python{major}.{minor}{exe_suffix}'.format(
-            major=python_version.major,
-            minor=python_version.minor,
-            exe_suffix=get_exe_suffix()),
-                  python_version=python_version)
+        if python_version.minor:
```
Comment (vereszol) [inline src/virtualenvrunner/pythonversionrun.py]: Instead of `if .. else` it would be better to use a helper function, same way like `get_exe_suffix`.

Even better solution would be to treat version as one entity and implement its string representation in PythonVersion class, something like below:

```
class PythonVersion:
    def __init__(self, major, minor):
        self.major = major
        self.minor = minor

    def __str__(self):
        if not self.minor:
            return str(self.major)
        return '{}.{}'.format(self.major, self.minor)

def get_python_versions():
    for v in [(2, None), (2, 7), (3, None), (3, 4), (3, 5), (3, 6)]:
        yield PythonVersion(*v)

for v in get_python_versions():
    print('python{version}'.format(version=v))
```

File: src/virtualenvrunner/pythonversionrun.py [hunk 2]
```diff
@@ -20,16 +20,22 @@ def _get_run_lambda(self, python_version):
...
+                major=python_version.major,
+                exe_suffix=get_exe_suffix()),
+                python_version=python_version)
 
     def __getattr__(self, name):
         try:
             return self._run_functions[name]
         except KeyError:
-            raise AttributeError('{cls} object has no attribute {name}'.format(
+            raise AttributeError('{cls} object has no attribute {name}'.format(  # pylint: disable=raise-missing-from
```
Comment (vereszol) [inline src/virtualenvrunner/pythonversionrun.py:32]: Disabling pylint checks `raise-missing-from` and `super-with-arguments` should be on its own commit, as it is not related to content of this change.

File: src/virtualenvrunner/pythonversionrun.py (hunk 1 above)
Comment (beleki) [reply src/virtualenvrunner/pythonversionrun.py]: Refactored

File: src/virtualenvrunner/pythonversionrun.py (hunk 2 above)
Comment (beleki) [reply src/virtualenvrunner/pythonversionrun.py:32]: Added as new commit

File: src/virtualenvrunner/python_versions.py [hunk 3]
```diff
@@ -15,4 +15,3 @@ def __str__(self):
 def get_python_versions():
     for v in [(2, None), (2, 7), (3, None), (3, 4), (3, 5), (3, 6)]:
         yield PythonVersion(*v)
-
```
Comment (vereszol) [inline src/virtualenvrunner/python_versions.py]: Trailing newline should not be removed

File: setup.py [hunk 4]
```diff
@@ -50,9 +50,9 @@ def get_console_scripts():
     return (
         ['{clif.cli} = virtualenvrunner.cli:{clif.function}'.format(
             clif=clif) for clif in get_clifunctions()] +
-        [('{clif.cli}{major}.{minor} = '
+        [('{clif.cli}{version} = '
           'virtualenvrunner.cli:{clif.function}{major}{minor}'.format(
-              major=v.major, minor=v.minor, clif=clif))
+             version=str(v), major=v.major, minor=v.minor, clif=clif))
```
Comment (vereszol) [inline setup.py]: No need to use `str()` as `format` is calling that automatically unless some format is specified.

File: src/virtualenvrunner/pythonversionrun.py [hunk 5]
```diff
@@ -20,11 +20,10 @@ def _get_run_lambda(self, python_version):
         return lambda: self._run_with_python_version(python_version)
 
     def _run_with_python_version(self, python_version):
-        self._run(pythonexe='python{major}.{minor}{exe_suffix}'.format(
-            major=python_version.major,
-            minor=python_version.minor,
+        self._run(pythonexe='python{version}{exe_suffix}'.format(
+            version=str(python_version),
```
Comment (vereszol) [inline src/virtualenvrunner/pythonversionrun.py]: No need to use `str`

File: tox.ini [hunk 6]
```diff
@@ -1,15 +1,15 @@
 # Copyright (C) 2019, Nokia
 
 [tox]
-envlist = py27, py36, docs, pylint
```
Comment (vereszol) [inline tox.ini]: Should not remove `py36` as it seems to be still supported by the tool and we even provide `run_in_virtualenv3.6`.
On the other hand, adding `py37` is fine, we could even add it to `get_python_versions`.

File: tox.ini [hunk 7]
```diff
@@ -1,15 +1,15 @@
...
+envlist = py27, py37, docs, pylint
 
 [base]
 deps =
     mock
     pytest
     pytest-cov
     pytest-pep8
-    pytest-console-scripts
+    pytest-console-scripts==0.1.6
```
Comment (vereszol) [inline tox.ini:11]: Reason for changes to dependencies (and also pytest options) should be explained in commit message.
This (and pytest option changes) could be in same commit where pylint checks are disabled.

File: tox.ini [hunk 8]
```diff
@@ -31,8 +31,6 @@ commands = {posargs:py.test -v\
            {envsitepackagesdir}/virtualenvrunner {toxinidir}/tests}
 
 [pytest]
-addopts = --pep8
```
Comment (vereszol) [inline tox.ini:34]: If pep8 checks are not used any more you could remove `pytest-pep8` from dependency list.

File: tox.ini [hunk 9]
```diff
@@ -31,8 +31,6 @@ commands = {posargs:py.test -v\
...
-pep8maxlinelength = 90
```
Comment (vereszol) [inline tox.ini:35]: As pep8 is not used any more pylint should be configured to expect same line length at [.pylintrc#L196](https://github.com/nokia/virtualenvrunner/blob/master/.pylintrc#L196)

File: src/virtualenvrunner/pythonversionrun.py (hunk 2 above)
Comment (vereszol) [reply src/virtualenvrunner/pythonversionrun.py:32]: OK

File: tox.ini (hunk 6 above)
Comment (vereszol) [reply tox.ini]: OK

File: tox.ini (hunk 7 above)
Comment (vereszol) [reply tox.ini:11]: ok

File: tox.ini (hunk 8 above)
Comment (vereszol) [reply tox.ini:34]: ok

File: tox.ini (hunk 9 above)
Comment (vereszol) [reply tox.ini:35]: ok
